networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -147,30 +150,61 @@ where
 			}
 		}
 
-		let length = transactions.len();
-		if length > 0 {
-			let mut light_node_client = self.light_node_client.clone();
-			let span = info_span!(
+		if transactions.len() > 0 {
+			info!(
 				target: "movement_timing",
-				"batch_write",
 				batch_id = %batch_id,
-				length = length
+				transaction_count = transactions.len(),
+				"built_batch_write"
 			);
-			light_node_client
-				.batch_write(BatchWriteRequest { blobs: transactions })
-				.instrument(span)
-				.await?;
-			// We now consider the transactions no longer in mempool flight.
-			self.executor.decrement_transactions_in_flight(length as u64);
+			let batch_write = BatchWriteRequest { blobs: transactions };
+			sender.send(batch_write).await?;
 		}
 
 		Ok(())
 	}
 
+	pub async fn send_transaction_batch_writes(
+		&self,
+		sender: Sender<BatchWriteRequest>,
+	) -> Result<(), anyhow::Error> {
+		loop {
+			self.next_transaction_batch_write(sender.clone()).await?;
+		}
+	}
+
+	pub async fn submit_transaction_batch_writes(
+		&self,
+		receiver: Receiver<BatchWriteRequest>,
+	) -> Result<(), anyhow::Error> {
+		loop {
+			let batch_write = receiver.recv().await?;
+			let mut light_node_client = self.light_node_client.clone();
+			// batch_writes can be submitted in parallel without waiting for the previous one to finish
+			tokio::spawn(async move {
+				light_node_client.batch_write(batch_write).await?;
+				Ok::<(), anyhow::Error>(())
+			});
+		}
+	}
+
 	pub async fn write_transactions_to_da(&self) -> Result<(), anyhow::Error> {
 		loop {
-			self.tick_write_transactions_to_da().await?;
+			let (batch_sender, batch_receiver) = async_channel::unbounded();
```

Comment:
Try replacing `async_channel` with `tokio::sync::mpsc`. However here, it's dubious we need a channel at all yet, see below.

Reply:
Yeah, should be `mpsc` not `mpmc`.